hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/HoodieSparkKryoRegistrar.scala

```diff
@@ -44,12 +45,15 @@ import org.apache.spark.serializer.KryoRegistrator
  * </ol>
  */
 class HoodieSparkKryoRegistrar extends HoodieCommonKryoRegistrar with KryoRegistrator {
+
   override def registerClasses(kryo: Kryo): Unit = {
     ///////////////////////////////////////////////////////////////////////////
     // NOTE: DO NOT REORDER REGISTRATIONS
     ///////////////////////////////////////////////////////////////////////////
     super[HoodieCommonKryoRegistrar].registerClasses(kryo)
 
+    kryo.register(classOf[HoodieKey], new HoodieKeySerializer)
```

Comment:
so we need this because we still want to be able to make use of serde optimization between shuffles?

Reply:
I guess this ONLY applied to the shuffle and not for the actual serialization? i,e this instance of kryo is not a shared/singleton or sth. correcT?

Reply:
@codope correct. This is still needed to make sure we're not shuffling fully-qualified class names in the payload

@vinothchandar correct. This registrar is only used on Spark's instance of Kryo used for shuffling